sources/kube_factory.go

```diff
@@ -31,11 +32,13 @@ import (
 )
 
 const (
-	defaultApiVersion     = "v1beta3"
-	defaultInsecure       = false
-	defaultKubeletPort    = 10255
-	defaultKubeConfigFile = "/etc/kubernetes/kubeconfig/kubeconfig"
-	defaultKubeletHttps   = false
+	defaultApiVersion         = "v1beta3"
+	defaultInsecure           = false
+	defaultKubeletPort        = 10255
+	defaultKubeConfigFile     = "/etc/kubernetes/kubeconfig/kubeconfig"
+	defaultKubeletHttps       = false
+	defaultUseServiceAccount  = false
+	defaultServiceAccountFile = "/var/run/secrets/kubernetes.io/serviceaccount/token"
```

Comment:
Yeah that's how service accounts work. You create a service account, a token is generated, then you can reference that service account in your pod spec & token is automounted.